kythe/cxx/common/kzip_writer.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef KYTHE_CXX_COMMON_KZIP_WRITER_H_
+#define KYTHE_CXX_COMMON_KZIP_WRITER_H_
+
+#include <zip.h>
+#include <vector>
+
+#include "kythe/cxx/common/index_writer.h"
+#include "kythe/cxx/common/status_or.h"
+
+namespace kythe {
+
+class KzipWriter : public IndexWriterInterface {
```

Comment:
Please document this—in particular the requirement that the caller arrange to call Close and check its result before giving up on the object.